docs: Add documentation for s390x cpu implementation

dilipgb opened the pull request:
This PR provides documentation for building  vLLM cpu implementation for s390x architecture. PR for s390x cpu implementation is approved and awaiting merge. https://github.com/vllm-project/vllm/pull/12613

dilipgb committed:
docs: Documentation for s390x cpu builds

Signed-off-by: Dilip Gowda Bhagavan <dilip.bhagavan@ibm.com>

dilipgb committed:
chore: cleanup contents

Signed-off-by: Dilip Gowda Bhagavan <dilip.bhagavan@ibm.com>

dilipgb committed:
chore: correct the typo in s390x docs

Signed-off-by: Dilip Gowda Bhagavan <dilip.bhagavan@ibm.com>

hmellor reviewed (approved):
From a docs perspective, this looks good to me.

Only suggestion is to make sure `torchvision` doesn't have a space.

dilipgb committed:
Apply suggestions from code review

Co-authored-by: Harry Mellor <19981378+hmellor@users.noreply.github.com>
Signed-off-by: Dilip Gowda Bhagavan <dilip.bhagavan@ibm.com>

dilipgb committed:
resolving merge conflicts in installation file

Signed-off-by: Dilip Gowda Bhagavan <dilip.bhagavan@ibm.com>

dilipgb committed:
fix:adapt to document structure changes post rebase

Signed-off-by: Dilip Gowda Bhagavan <dilip.bhagavan@ibm.com>

dilipgb commented:
@hmellor, I have rebased the PR and updated installation.md with path changes due to rebase. Also addressed whitespace in torchvision. Can we merge the PR, if this looks good?
